internal/agent/device/console/incoming_streams.go

```diff
@@ -38,6 +45,7 @@ func newIncomingStreams(streamClient grpc_v1.RouterService_StreamClient, stdin i
 
 func (i *incomingStreams) run(wg *sync.WaitGroup) {
```

Comment:
if run was context-aware you could also ensure shutdown when the parent context is cancelled 

```go
      for {
		select {
		case <-ctx.Done():
			i.log.Debug("incoming streams cancelled")
			return
```

Reply:
The purpose of it is to call cancel when a request to close the stream is received by gRPC at line 59

Reply:
right I just mean lets say the agent process exits shutdown will cancel the parent context. maybe this is already covered just an observation.

Reply:
Right, but the purpose of this change was to propagate connection closure which is signaled from the api-server.  Until now, it was not handled properly.